### PR TITLE
subsys: nffs_fs: Automatically calculate the area size

### PR DESCRIPTION
**Pull request by findlayfeng**
In some cases, the capacity of the flash chip is not deterministic.
In this case, if nffs uses the last area of the chip, size can be set to 0.
Try to calculate area size automatically.

Signed-off-by: Findlay Feng <i@fengch.me>

### CODE REVIEW

File: subsys/fs/nffs_fs.c [hunk 1]
```diff
@@ -524,6 +524,17 @@ static int nffs_rename(struct fs_mount_t *mountp, const char *from,
 	return translate_error(rc);
 }
 
+static bool get_flash_size_cb(const struct flash_pages_info *info, void *data)
+{
+	struct nffs_flash_desc *flash_desc = data;
+
+	if (flash_desc->area_offset + flash_desc->area_size == info->start_offset) {
+		flash_desc->area_size += info->size;
```
Comment (nvlsianpu) [inline subsys/fs/nffs_fs.c]: why not '=' instead?

After this line;
this condition should be valid exactly for the one flash sector - so use `return false' for  finish the iteration once this happened,

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: subsys/fs/nffs_fs.c (hunk 1 above)
Comment (findlayfeng) [reply subsys/fs/nffs_fs.c]: one? why not all?

Comment (findlayfeng) [reply subsys/fs/nffs_fs.c]: Here is the use of all the remaining pages, not just one page.

File: subsys/fs/nffs_fs.c [hunk 2]
```diff
@@ -524,6 +524,17 @@ static int nffs_rename(struct fs_mount_t *mountp, const char *from,
 	return translate_error(rc);
 }
 
+static bool get_flash_size_cb(const struct flash_pages_info *info, void *data)
+{
+	struct nffs_flash_desc *flash_desc = data;
+
+	if (flash_desc->area_offset + flash_desc->area_size == info->start_offset) {
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply subsys/fs/nffs_fs.c]: I see now that it works as you describes - so it calculates the continuous area size right after FLASH_AREA_STORAGE_OFFSE.

**Comment by nvlsianpu** [comment]
@sjanc @andrzej-kaczmarek can one of you take a look at this as well?

**Comment by ioannisg** [comment]
@findlayfeng any plans to continue working on this one?

**Pull request closed without merging**